### PR TITLE
Add new spec validations on the p2p layer in Bellatrix

### PR DESCRIPTION
**Pull request by jmozah**
This PR adds a new validation where in the peer is not penalised (p2p peer scoring) if the blocks parent is optimistic.

### CODE REVIEW

File: beacon-chain/blockchain/testing/mock.go [hunk 1]
```diff
@@ -448,5 +448,5 @@ func (s *ChainService) IsOptimistic(_ context.Context) (bool, error) {
 
 // IsOptimisticForRoot mocks the same method in the chain service.
 func (s *ChainService) IsOptimisticForRoot(_ context.Context, _ [32]byte, _ types.Slot) (bool, error) {
-	return false, nil
+	return s.Optimistic, nil
```
Comment (terencechain) [inline beacon-chain/blockchain/testing/mock.go:451]: Let's return false with todo and target this for `develop`. See:
https://github.com/prysmaticlabs/prysm/pull/10276

Comment (prestonvanloon) [reply beacon-chain/blockchain/testing/mock.go:451]: This is a mock

File: beacon-chain/sync/validate_beacon_blocks.go [hunk 2]
```diff
@@ -27,6 +27,10 @@ import (
 	"go.opencensus.io/trace"
 )
 
+var (
+	ErrOptimisticParent = errors.New("parent of the block is optimistic")
```
Comment (prestonvanloon) [inline beacon-chain/sync/validate_beacon_blocks.go:31]: Please add a godoc comment for all public things.

File: beacon-chain/blockchain/testing/mock.go (hunk 1 above)
Comment (jmozah) [reply beacon-chain/blockchain/testing/mock.go:451]: Yeap.. making it return `s.Optimistic` makes it easy to test.

File: beacon-chain/sync/validate_beacon_blocks.go (hunk 2 above)
Comment (jmozah) [reply beacon-chain/sync/validate_beacon_blocks.go:31]: aah.. it need not be public. But still i added comment to make it easy to understand.

**Review by zapuly76 — APPROVED** [review]

**Review by zapuly76 — DISMISSED** [review]

**Comment by terencechain** [comment]
same here, please fix the diff, thank you!

**Comment by jmozah** [comment]
Closing in favour of https://github.com/prysmaticlabs/prysm/pull/10301

**Pull request closed without merging**